### ISSUE TITLE
Bug in Download Response headers 

### ISSUE THREAD

**Issue by mansai** [proudmonkey/AutoWrapper#55]
[HttpPost]        
        [AutoWrapIgnore]
        public async Task< FileStreamResult> GetExcel()
        {
            var filePath = "xxx..xlsx";
            var file = System.IO.File.OpenRead(filePath);
            var provider = new FileExtensionContentTypeProvider();           
            var fileInfo = new System.IO.FileInfo(filePath);           
            var memi = provider.Mappings[fileInfo.Extension];
            return File(file, memi, fileInfo.Name);
        }
Bug:
content-type: application/json 
![{978E9BF7-6A16-49EE-938F-0BB2E03FEDA6}_20200601115104](https://user-images.githubusercontent.com/3213405/83374633-c7f58b00-a3fe-11ea-8e24-01d9d337f374.jpg)

Right:
content-type: application/vnd.openxmlformats-officedocument.spreadsheetml.sheet

**Comment by hueifeng** [comment]
@mansai I think it should be the following
```
     [HttpGet]
        [AutoWrapIgnore]
        public async Task<FileStreamResult> GetExcel()
        {
            var filePath = "test.txt";
            var file = System.IO.File.OpenRead(filePath);
            var provider = new FileExtensionContentTypeProvider();
            var fileInfo = new System.IO.FileInfo(filePath);
            var memi = provider.Mappings[fileInfo.Extension];
            return File(file, memi, fileInfo.Name);
        }
```

**Comment by mansai** [comment]
![image](https://user-images.githubusercontent.com/3213405/83379572-71904880-a40e-11ea-96f0-03b49781bfbd.png)

**Comment by mansai** [comment]
![1](https://user-images.githubusercontent.com/3213405/83380222-fcbe0e00-a40f-11ea-8d45-d33061af7fed.jpg)
![2](https://user-images.githubusercontent.com/3213405/83380229-02b3ef00-a410-11ea-8f0c-4eb70e2dc0aa.jpg)
![3](https://user-images.githubusercontent.com/3213405/83380237-08113980-a410-11ea-9204-54bbca4f65c3.jpg)

**Comment by hueifeng** [comment]
@mansai Ok, I have reproduced the problem, you can first use the following code snippet to return your expected result.
```
     app.UseApiResponseAndExceptionWrapper(new AutoWrapperOptions
            {
                IsApiOnly = false,
            });
```

**Comment by mansai** [comment]
There are still problems
![5](https://user-images.githubusercontent.com/3213405/83387958-31d25c80-a420-11ea-8f19-6d0a29c0d588.jpg)

**Comment by hueifeng** [comment]
@mansai I'm sorry I can't reproduce your results. I do n’t know if you have executed `dotnet build`, or to better solve the problem, I suggest you provide a reproducible demo

**Comment by mansai** [comment]
[DownLoadDemo.zip](https://github.com/proudmonkey/AutoWrapper/files/4710337/DownLoadDemo.zip)
![6](https://user-images.githubusercontent.com/3213405/83393040-b6c17400-a428-11ea-85d1-7e7b708e577b.jpg)
![7](https://user-images.githubusercontent.com/3213405/83393223-ff792d00-a428-11ea-8a82-91cc47f49d8c.jpg)

![8](https://user-images.githubusercontent.com/3213405/83393348-364f4300-a429-11ea-9515-7309515de638.jpg)

**Comment by proudmonkey** [comment]
@mansai Thanks for the feedback! I appreciate it. This should now be resolved with the latest release here: https://www.nuget.org/packages/AutoWrapper.Core/4.2.0

Sample usage:

```csharp
        [HttpGet("getfile")]
        [AutoWrapIgnore]
        public FileStreamResult GetFile2()
        {
            var filePath = $"{Environment.CurrentDirectory}/Logs/log_20200501.txt";
            var file = System.IO.File.OpenRead(filePath);
            var provider = new FileExtensionContentTypeProvider();
            var fileInfo = new System.IO.FileInfo(filePath);
            var memi = provider.Mappings[fileInfo.Extension];
            return File(file, memi, fileInfo.Name);
        }

        [HttpGet("getfile2")]
        [AutoWrapIgnore]
        public FileStreamResult GetFile()
        {
            var path = $"{Environment.CurrentDirectory}/Logs/log_20200501.txt";
            var stream = System.IO.File.OpenRead(path);
            return new FileStreamResult(stream, "application/octet-stream");
        }

        [HttpGet("getpdf")]
        [AutoWrapIgnore]
        public IActionResult GetPdf()
        {
            var path = $"{Environment.CurrentDirectory}/sample.pdf";
            byte[] fileBytes = System.IO.File.ReadAllBytes(path);
            return new FileContentResult(fileBytes, "application/pdf");
        }

        [HttpGet("getpdf2")]
        [AutoWrapIgnore]
        public IActionResult GetPdf2()
        {
            var path = $"{Environment.CurrentDirectory}/sample.pdf";
            var stream = new FileStream(path, FileMode.Open);
            return new FileStreamResult(stream, "application/pdf");
        }

        //returns text/plain
        [HttpGet("getstring")]
        [AutoWrapIgnore]
        public ContentResult GetString()
        {
            return Content("Hello");
        }
```

**Comment by mansai** [comment]
txt file download ok，Unable to open Excel or pdf，content-length changed

**Comment by proudmonkey** [comment]
@mansai  

I've released v4.2.2 here: https://www.nuget.org/packages/AutoWrapper.Core/4.2.2

Please update to that and it should fix the issue.

Thanks!
